montepy/input_parser/syntax_node.py

```diff
@@ -295,26 +304,117 @@ class GeometryTree(SyntaxNodeBase):
     :type right: GeometryTree, ValueNode
     """
 
-    def __init__(self, name, tokens, op, left, right=None):
+    def __init__(
+        self,
+        name,
+        tokens,
+        op,
+        left,
+        right=None,
+        left_short_type=None,
+        right_short_type=None,
+    ):
         super().__init__(name)
         assert all(list(map(lambda v: isinstance(v, SyntaxNodeBase), tokens.values())))
         self._nodes = tokens
         self._operator = Operator(op)
         self._left_side = left
         self._right_side = right
+        self._left_short_type = left_short_type
+        self._right_short_type = right_short_type
 
     def __str__(self):
-        return f"Geometry: ( {self._left_side} {self._operator} {self._right_side} )"
+        return (
+            f"Geometry: ( {self._left_side}"
+            f" {f'Short:{self._left_short_type.value}' if self._left_short_type else ''}"
+            f" {self._operator} {self._right_side} "
+            f"{f'Short:{self._right_short_type.value}' if self._right_short_type else ''})"
+        )
 
     def __repr__(self):
         return str(self)
 
     def format(self):
+        if self._left_short_type or self._right_short_type:
+            return self._format_shortcut()
         ret = ""
         for node in self.nodes.values():
             ret += node.format()
         return ret
 
+    def mark_last_leaf_shortcut(self, short_type):
+        if self.right is not None:
+            node = self.right
+            if self._right_short_type:
+                return
+        else:
+            node = self.left
+            if self._left_short_type:
+                return
+        if isinstance(node, type(self)):
+            return node.mark_last_leaf_shortcut(short_type)
+        if self.right is not None:
+            self._right_short_type = short_type
+        else:
+            self._left_short_type = short_type
+
+    def _flatten_shortcut(self):
+        def add_leaf(list_node, leaf, short_type):
+            end = list_node.nodes[-1] if len(list_node) > 0 else None
+
+            def flush_shortcut():
+                end.load_nodes(end.nodes)
+
+            def start_shortcut():
+                short = ShortcutNode(short_type=short_type)
+                # give an interpolate it's old beginning to give it right
+                # start value
+                if short_type in {
+                    Shortcuts.LOG_INTERPOLATE,
+                    Shortcuts.INTERPOLATE,
+                } and isinstance(end, ShortcutNode):
+                    short.append(end.nodes[-1])
+                    short._has_pseudo_start = True
```

Comment:
Can't decide if this flag is hacky or not.

It is necessary to update `self._begin` with `load_nodes`.

Reply:
a little bit but it's ok